### PR TITLE
Aussie seaweed/ltc6810

### PR DESCRIPTION
**Pull request by AussieSeaweed**

### CODE REVIEW

File: iclib/ltc6810.py [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+                command_bytes,
+                cls.get_packet_error_code_bytes(command_bytes),
+                poll_data_bytes,
+            ),
+        )
+
+    @classmethod
+    def get_address_write_command_bytes(
+            cls,
+            address: int,
```
Comment (Michael12304) [inline iclib/ltc6810.py:143]: Is including unused param "address" here intentional (i.e. to maintain consistent formatting for command functions?)

File: iclib/ltc6810.py [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+        - ``0b000``: all cells.
+        - ``0b001``: Cell 1.
+        - ``0b010``: Cell 2.
+        - ``0b011``: Cell 3.
+        - ``0b100``: Cell 4.
+        - ``0b101``: Cell 5.
+        - ``0b110``: Cell 6.
+
+        :param ch_mode: The ADC mode.
+        :param DCP: Dischrage permitted.
```
Comment (Michael12304) [inline iclib/ltc6810.py:285]:
```suggestion
        :param DCP: Discharge permitted.
```

File: iclib/ltc6810.py [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+
+    def ADCV(
+            self,
+            ch_mode: CHMode,
+            DCP: bool,
+            CH: int,
+            address: int | None = None,
+            sleep: bool = True,
+    ) -> None:
+        """Start cell voltage ADC conversion and poll status.
```
Comment (Michael12304) [inline iclib/ltc6810.py:267]:
```suggestion
        """Start cell voltage ADC conversion and poll status. Refer to Table 40 (Page 61) of datasheet.
```

File: iclib/ltc6810.py [hunk 4]
```diff
@@ -0,0 +1,622 @@
...
+    @dataclass
+    class CVAR:
+        """Cell voltage register group A."""
+
+        C1V: float
+        C2V: float
+        C3V: float
+
+    def RDCVA(self, address: int | None = None) -> CVAR:
+        """Read cell voltage register group A.
```
Comment (Michael12304) [inline iclib/ltc6810.py:324]:
```suggestion
        """Read cell voltage register group A. Refer to Table 40 (Page 61) on datasheet.
```

File: iclib/ltc6810.py [hunk 5]
```diff
@@ -0,0 +1,622 @@
...
+    @dataclass
+    class CVBR:
+        """Cell voltage register group B."""
+
+        C4V: float
+        C5V: float
+        C6V: float
+
+    def RDCVB(self, address: int | None = None) -> CVBR:
+        """Read cell voltage register group B.
```
Comment (Michael12304) [inline iclib/ltc6810.py:357]:
```suggestion
        """Read cell voltage register group B. Refer to Table 40 (Page 61) on datasheet.
```

File: iclib/ltc6810.py [hunk 6]
```diff
@@ -0,0 +1,622 @@
...
+        """27 kHz Mode (Fast)."""
+        M14000 = 0b01, 695e-6, 229e-6
+        """14 kHz Mode."""
+        M7000 = 0b10, 1.2e-3, 403e-6
+        """7 kHz Mode (Normal)."""
+        M3000 = 0b10, 1.9e-3, 520e-6
+        """3 kHz Mode."""
+        M2000 = 0b11, 3.3e-3, 752e-6
+        """2 kHz Mode."""
+        M1000 = 0b00, 6.1e-3, 1.2e-3
```
Comment (Michael12304) [inline iclib/ltc6810.py:394]: Datasheet says 6.0ms for 1khz, not 6.1ms

File: iclib/ltc6810.py [hunk 7]
```diff
@@ -0,0 +1,622 @@
...
+
+    def AXOW(
+            self,
+            chg_mode: CHGMode,
+            PUP: bool,
+            CHG: int,
+            address: int | None = None,
+            sleep: bool = True,
+    ) -> None:
+        """Start GPIOS/Cell 0/REF2 ADC open wire conversion.
```
Comment (Michael12304) [inline iclib/ltc6810.py:423]:
```suggestion
        """Start GPIOS/Cell 0/REF2 ADC open wire conversion. Refer to Table 40 (Page 62) on datasheet.
```

File: iclib/ltc6810.py [hunk 8]
```diff
@@ -0,0 +1,622 @@
...
+            _sleep(timeout)
+
+    @dataclass
+    class AVAR:
+        S0V: float
+        G1V: float
+        G2V: float
+
+    def RDAUXA(self, address: int | None = None) -> AVAR:
+        """Read auxiliary register group A.
```
Comment (Michael12304) [inline iclib/ltc6810.py:478]:
```suggestion
        """Read auxiliary register group A. Refer to Table 40 (Page 61) on datasheet.
```

File: iclib/ltc6810.py [hunk 9]
```diff
@@ -0,0 +1,622 @@
...
+    @dataclass
+    class AVBR:
+        """Auxiliary register group B."""
+
+        G3V: float
+        G4V: float
+        REF: float
+
+    def RDAUXB(self, address: int | None = None) -> AVBR:
+        """Read auxiliary register group B.
```
Comment (Michael12304) [inline iclib/ltc6810.py:511]:
```suggestion
        """Read auxiliary register group B. Refer to Table 40 (Page 61) on datasheet.
```

File: iclib/ltc6810.py [hunk 10]
```diff
@@ -0,0 +1,622 @@
...
+                    self.FDRF,
+                    self.DIS_RED,
+                    self.DTMEN,
+                )
+            )
+
+            return CFGR0, CFGR1, CFGR2, CFGR3, CFGR4, CFGR5
+
+    def WRCFG(self, CFGR: CFGR, address: int | None = None) -> None:
+        """Write configuration register group.
```
Comment (Michael12304) [inline iclib/ltc6810.py:607]:
```suggestion
        """Write configuration register group. Refer to Table 40 (Page 61) on datasheet.
```

File: iclib/ltc6810.py (hunk 1 above)
Comment (AussieSeaweed) [reply iclib/ltc6810.py:143]: Good catch, it's a mistake. I am supposed to use it in the code.

File: iclib/ltc6810.py (hunk 6 above)
Comment (AussieSeaweed) [reply iclib/ltc6810.py:394]: Yup, good catch!

**Pull request merged**